src/client.js

```diff
@@ -248,6 +248,13 @@ module.exports = class IrcClient extends EventEmitter {
             }
         });
 
+        commands.on('user mode', function(event) {
+            if (client.user.nick !== event.nick) {
+                return; // This should never happen
```

Comment:
can someone more knowledgeable than me please sanity check if that's true?
Not sure how the sequence ends up when we do `MODE <nick>`, followed by a nick change.

I assumed that the response would be sequential, meaning we'd first get the RPL_UMODEIS back prior to the nick change event.

If that's not true we might want to skip that sanity check and just rely on the server to not do dumb things